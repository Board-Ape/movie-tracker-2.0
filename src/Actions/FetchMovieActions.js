export default function FetchMovieActions(url) {
  return dispatch => {

    dispatch({ type: 'MOVIES_IS_LOADING' });

    return fetch(`${url}`)
      .then(res => res.json())
      .then(movies => { 
        return dispatch({ type: 'MOVIES_FETCH_DATA_SUCCESS', payload: movies });
      })
      .catch( () => {
        return dispatch({ type: 'MOVIES_HAS_ERRORED' });
      });

  };
}