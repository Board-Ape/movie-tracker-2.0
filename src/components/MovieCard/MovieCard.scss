@import '../../movietrackericons/css/movietrackericons.css';

.outer-card-container {
   margin: 10px;
   width: 290px;
   height: 500px;
   perspective: 1000;
   z-index: 1;
    
    .inner-card-container {
      width: 100%;
      height: 100%;
      transform-style: preserve-3d;
      transition: all 0.8s linear;
      .face {
          position: absolute;
          width: 100%;
          height: 100%;
          backface-visibility: hidden;
          display: flex;
          flex-wrap: wrap;
          box-shadow: 2px 2px 2px 2px rgb(55, 55, 55);
          border-radius: 10px;
          border: 1px solid black;
          img {
            width: 290px;
            height: 430px;
            border-radius: 10px 10px 0px 0px;
          
          }
          .bottom-container {
            display: flex;
            justify-content: space-between;
            height: 70px;
            background-color: white;
            width: 100%;
            border-top: 1px solid black;
            border-radius: 0px 0px 10px 10px;
        
            h1 {
              font-size: 25px;
              margin: 0px;
              height: 70px;
              width: 170px;
              display: flex;
              justify-content: center;
              align-items: center;
              text-align: center;
            }
            
            h1.favorited {
              background-color: #00AEFF;
              color: black;
            }
        
            button{
              height: 70px;
              width: 60px;
              border: none;
              cursor: pointer;
              font-size: 18px;
              padding: 0px;
              &:hover {
                background-color: black;
                color: #00AEFF;

              }
              i {
                font-size: 30px;
              }
              &:nth-child(1) {
                border-right: 1px solid black;
                border-radius: 0px 0px 0px 10px;
              }
              &:nth-child(3) {
                border-left: 1px solid black;
                border-radius: 0px 0px 10px 0px;
              }
            }
          }
      }
      .face.card-back {
          display: flex;
          box-sizing: border-box;
          transform: rotateY(180deg);
          height: 500px;
          width: 290px;
          background-color: darkgrey;;
          .top-section {
            height: 430px;
            .top-row {
              display: flex;
              justify-content: space-between;
              background-color: #00AEFF;
              border-radius: 10px 10px 0px 0px;
              height: 100px;
              h1 {
                width: 100%;
                font-size: 30px;
                margin: 0px;
                padding: 10px;
              }
              p {
                font-size: 50px;
                margin: 0px;
                padding-right: 10px;
              }
            }
            p {
              font-size: 20px;
              padding: 10px;
              margin: 0px;
              height: 310px;
              overflow: scroll;              
            }
          }
          .bottom-container {
            display: flex;
            justify-content: space-between;
            height: 70px;
            background-color: white;
            width: 100%;
            border-radius: 0px 0px 10px 10px;
            border-top: 1px solid black;
            h1 {
              font-size: 25px;
              margin: 0px;
              height: 70px;
              width: 170px;
              display: flex;
              justify-content: center;
              align-items: center;
              text-align: center;
            }
            h1.favorited {
              background-color: #00AEFF;
              color: black;
            }
        
            button{
              height: 70px;
              width: 60px;
              border: none;
              cursor: pointer;
              font-size: 18px;
              padding: 0px;
              &:hover {
                background-color: black;
                color: #00AEFF;

              }
              i {
                font-size: 30px;
              }
              &:nth-child(1) {
                border-right: 1px solid black;
                border-radius: 0px 0px 0px 10px;
              }
              &:nth-child(3) {
                border-left: 1px solid black;
                border-radius: 0px 0px 10px 0px;
              }
            }
          }
      }
    }
}

.outer-card-container.flip {
  .inner-card-container {
    transform: rotateY(180deg);
    box-shadow: -4px 4px 4px rgba(0,0,0,0.2);    
    
   }  
 }
