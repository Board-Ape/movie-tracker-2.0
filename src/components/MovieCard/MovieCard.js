import React, { Component } from 'react';
import './MovieCard.css';
import { connect } from 'react-redux';
import { addFavorite, removeFavorite } from '../../Actions';
import PropTypes from 'prop-types';

export class MovieCard extends Component{
  constructor() {
    super();
    this.state = {
      cardFlipped: false
    };
  }


  handleFavorite = (movie) => {
    const currentFavorites = this.props.favorites;
    const isFavorite = currentFavorites.find( movieId => {
      return movieId.title === movie.title;
    });

    if (!isFavorite) {
      this.props.addFavorite(
        movie,
        this.props.user.userData.id
      );
    } else {
      this.props.removeFavorite(
        movie,
        this.props.user.userData.id
      );
    }
  };

  checkFavorites = (movie) => {
    if (!this.props.user.signedIn) {
      alert('You must sign-in to add a favorite');
    } else {
      this.handleFavorite(movie);
    }
  };

  flipCard = () => {
    const cardFlipped = this.state.cardFlipped ? false : true;
    this.setState({
      cardFlipped
    });
  };

  render() {
    const flipClass = this.state.cardFlipped ? 'flip' : '';
    const isFavorite = this.props.favorites.find( movieId => {
      return movieId.title === this.props.movie.title;
    });
    const starClass = isFavorite ? 'icon-star' : 'icon-star-empty';
    const cardClass = isFavorite ? 'favorited' : '';

    return (
      <div className={`outer-card-container ${flipClass}`}>
        <div className="inner-card-container">
          <article className="card-front face">
            <img src={
              `https://image.tmdb.org/t/p/w500${this.props.poster}`
            } alt="movie poster" />
            <div className="bottom-container">
              <button onClick={(event) => {
                event.preventDefault();
                this.checkFavorites(this.props.movie);
              }}>Favorite<i className={starClass}></i></button>
              <h1 className={cardClass}>{this.props.title}</h1>
              <button onClick={
                () => { this.flipCard(); }
              }>More Info<i className="icon-exchange"></i></button>
            </div>
          </article>
          <article className="card-back face">
            <div className="top-section">
              <div className="top-row">
                <h1>{this.props.title}</h1>
                <p className="rating">{this.props.rating}</p>
              </div>
              <p>{this.props.overview}</p>
            </div>
            <div className="bottom-container">
              <button onClick={(event) => {
                event.preventDefault();
                this.checkFavorites(this.props.movie);
              }}>Favorite<i className={starClass}></i></button>
              <h1 className={cardClass}>{this.props.title}</h1>
              <button onClick={
                () => { this.flipCard(); }
              }>More Info<i className="icon-exchange"></i></button>
            </div>
          </article>
        </div>
      </div>
    );
  }
}

export const mapStateToProps = (state) => {
  return {
    user: state.SignIn,
    favorites: state.favorites
  };
};

export const mapDispatchToProps = (dispatch) => {
  return {
    addFavorite: (movie, id) =>
      dispatch(addFavorite(movie, id)),
    removeFavorite: (userId, movieId) =>
      dispatch(removeFavorite(userId, movieId))
  };
};

MovieCard.propTypes = {
  removeFavorite: PropTypes.func.isRequired,
  movie: PropTypes.object.isRequired,
  poster: PropTypes.string,
  title: PropTypes.string,
  rating: PropTypes.number,
  overview: PropTypes.string,
  user: PropTypes.object,
  favorites: PropTypes.array.isRequired,
  addFavorite: PropTypes.func.isRequired,
  removeFavorites: PropTypes.func
};

export default connect(mapStateToProps, mapDispatchToProps)(MovieCard);
