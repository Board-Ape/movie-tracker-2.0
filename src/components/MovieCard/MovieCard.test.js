import { shallow } from 'enzyme';
import React from 'react';
import { MovieCard, mapStateToProps, mapDispatchToProps } from './MovieCard';
import * as actions from '../../Actions';

describe('tests for MovieCard component', () => {
  describe('component tests', () => {
    let cardWrapper;
    let mockProps;

    beforeEach(() => {
      mockProps = {
        user: {
          error: false,
          signedIn: true,
          userData:{ name: "abc", email: "abc", id: 3 }
        },
        favorites: [],
        addFavorite: jest.fn(),
        removeFavorite: jest.fn(),
        movie: {
          adult: false,
          backdrop_path: "/askg3SMvhqEl4OL52YuvdtY40Yb.jpg",
          genre_ids: [ 10751, 16, 12 ],
          id: 354912,
          original_language: "en",
          original_title: "Coco",
          overview: "Despite his familyâ€™s baffling generations-old ban on music, Miguel dreams of becoming an accomplished musician like his idol, Ernesto de la Cruz. Desperate to prove his talent, Miguel finds himself in the stunning and colorful Land of the Dead following a mysterious chain of events. Along the way, he meets charming trickster Hector, and together, they set off on an extraordinary journey to unlock the real story behind Miguel's family history.",
          popularity: 566.134415,
          poster_path: "/eKi8dIrr8voobbaGzDpe8w0PVbC.jpg",
          release_date: "2017-10-27",
          title: "Coco",
          video: false,
          vote_average: 7.6,
          vote_count: 483
        }
      };
      cardWrapper = shallow(<MovieCard {...mockProps} />);

    });
    it('should mount with the correct elements', () => {
      const expectedImgLength = 1;
      const expectedH1Length = 3;
      const expectedButtonLength = 4;

      expect(cardWrapper.find('img').length).toEqual(expectedImgLength);
      expect(cardWrapper.find('h1').length).toEqual(expectedH1Length);
      expect(cardWrapper.find('button').length).toEqual(expectedButtonLength);
    });

    it('should match the snapshot', () => {
      expect(cardWrapper).toMatchSnapshot();
    });

    it('should call addFavorite when the movie is not already in favorites', () => {
      const expectedParams = [mockProps.movie, 3];

      cardWrapper.instance().checkFavorites(mockProps.movie);
      cardWrapper.update();
      expect(mockProps.addFavorite).toHaveBeenCalledWith(...expectedParams);
      expect(mockProps.removeFavorite.mock.calls.length).toEqual(0);
    });

    it('should call remove favorites when the movie is in favorites', () => {
      const newMockProps = Object.assign(
        {},
        mockProps,
        {favorites: [{title: "Coco"}]}
      );

      const expectedParams = [newMockProps.movie, 3];

      cardWrapper = shallow(<MovieCard {...newMockProps} />);

      cardWrapper.instance().checkFavorites(mockProps.movie);
      cardWrapper.update();

      expect(newMockProps.removeFavorite).toHaveBeenCalledWith(...expectedParams);
      expect(newMockProps.addFavorite.mock.calls.length).toEqual(0);
    });
    describe('mapStateToProps', () => {
      it('should get user info and favorites array from the store', ()=> {
        const mockStore = {
          SignIn: {
            error: false,
            signedIn: true,
            userData:{ name: "abc", email: "abc", id: 3 }
          },
          favorites: []
        };

        const result = mapStateToProps(mockStore);
        expect(result.user).toEqual(mockStore.SignIn);
        expect(result.favorites).toEqual(mockStore.favorites);
      });
    });

    describe('mapDispatchToProps', () => {
      it('should call dispatch when addFavorite is called', () => {
        const mockDispatch = jest.fn();
        const mockParams = [mockProps.movie, 3];
        const result = mapDispatchToProps(mockDispatch);

        result.addFavorite(...mockParams);
        expect(mockDispatch).toHaveBeenCalled();
      });

      it('should call dispatch when removeFavorite is called', () => {
        const mockDispatch = jest.fn();
        const mockParams = [3, 12345];
        const result = mapDispatchToProps(mockDispatch);

        result.removeFavorite(...mockParams);
        expect(mockDispatch).toHaveBeenCalled();
      });
    });

  });
});