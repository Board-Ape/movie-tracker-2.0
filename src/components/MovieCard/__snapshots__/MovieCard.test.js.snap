// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for MovieCard component component tests should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MovieCard
    addFavorite={[Function]}
    favorites={Array []}
    movie={
        Object {
            "adult": false,
            "backdrop_path": "/askg3SMvhqEl4OL52YuvdtY40Yb.jpg",
            "genre_ids": Array [
              10751,
              16,
              12,
            ],
            "id": 354912,
            "original_language": "en",
            "original_title": "Coco",
            "overview": "Despite his familyâ€™s baffling generations-old ban on music, Miguel dreams of becoming an accomplished musician like his idol, Ernesto de la Cruz. Desperate to prove his talent, Miguel finds himself in the stunning and colorful Land of the Dead following a mysterious chain of events. Along the way, he meets charming trickster Hector, and together, they set off on an extraordinary journey to unlock the real story behind Miguel's family history.",
            "popularity": 566.134415,
            "poster_path": "/eKi8dIrr8voobbaGzDpe8w0PVbC.jpg",
            "release_date": "2017-10-27",
            "title": "Coco",
            "video": false,
            "vote_average": 7.6,
            "vote_count": 483,
          }
    }
    removeFavorite={[Function]}
    user={
        Object {
            "error": false,
            "signedIn": true,
            "userData": Object {
              "email": "abc",
              "id": 3,
              "name": "abc",
            },
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="inner-card-container"
>
        <article
                className="card-front face"
        >
                <img
                        alt="movie poster"
                        src="https://image.tmdb.org/t/p/w500undefined"
                />
                <div
                        className="bottom-container"
                >
                        <button
                                onClick={[Function]}
                        >
                                Favorite
                                <i
                                        className="icon-star-empty"
                                />
                        </button>
                        <h1
                                className=""
                        />
                        <button
                                onClick={[Function]}
                        >
                                More Info
                                <i
                                        className="icon-exchange"
                                />
                        </button>
                </div>
        </article>
        <article
                className="card-back face"
        >
                <div
                        className="top-section"
                >
                        <div
                                className="top-row"
                        >
                                <h1 />
                                <p
                                        className="rating"
                                />
                        </div>
                        <p />
                </div>
                <div
                        className="bottom-container"
                >
                        <button
                                onClick={[Function]}
                        >
                                Favorite
                                <i
                                        className="icon-star-empty"
                                />
                        </button>
                        <h1
                                className=""
                        />
                        <button
                                onClick={[Function]}
                        >
                                More Info
                                <i
                                        className="icon-exchange"
                                />
                        </button>
                </div>
        </article>
</div>,
      "className": "outer-card-container ",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <article
            className="card-front face"
>
            <img
                        alt="movie poster"
                        src="https://image.tmdb.org/t/p/w500undefined"
            />
            <div
                        className="bottom-container"
            >
                        <button
                                    onClick={[Function]}
                        >
                                    Favorite
                                    <i
                                                className="icon-star-empty"
                                    />
                        </button>
                        <h1
                                    className=""
                        />
                        <button
                                    onClick={[Function]}
                        >
                                    More Info
                                    <i
                                                className="icon-exchange"
                                    />
                        </button>
            </div>
</article>,
          <article
            className="card-back face"
>
            <div
                        className="top-section"
            >
                        <div
                                    className="top-row"
                        >
                                    <h1 />
                                    <p
                                                className="rating"
                                    />
                        </div>
                        <p />
            </div>
            <div
                        className="bottom-container"
            >
                        <button
                                    onClick={[Function]}
                        >
                                    Favorite
                                    <i
                                                className="icon-star-empty"
                                    />
                        </button>
                        <h1
                                    className=""
                        />
                        <button
                                    onClick={[Function]}
                        >
                                    More Info
                                    <i
                                                className="icon-exchange"
                                    />
                        </button>
            </div>
</article>,
        ],
        "className": "inner-card-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt="movie poster"
                src="https://image.tmdb.org/t/p/w500undefined"
/>,
              <div
                className="bottom-container"
>
                <button
                                onClick={[Function]}
                >
                                Favorite
                                <i
                                                className="icon-star-empty"
                                />
                </button>
                <h1
                                className=""
                />
                <button
                                onClick={[Function]}
                >
                                More Info
                                <i
                                                className="icon-exchange"
                                />
                </button>
</div>,
            ],
            "className": "card-front face",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "movie poster",
                "src": "https://image.tmdb.org/t/p/w500undefined",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <button
                    onClick={[Function]}
>
                    Favorite
                    <i
                                        className="icon-star-empty"
                    />
</button>,
                  <h1
                    className=""
/>,
                  <button
                    onClick={[Function]}
>
                    More Info
                    <i
                                        className="icon-exchange"
                    />
</button>,
                ],
                "className": "bottom-container",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Favorite",
                      <i
                        className="icon-star-empty"
/>,
                    ],
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Favorite",
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "className": "icon-star-empty",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "i",
                    },
                  ],
                  "type": "button",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                    "className": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "h1",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "More Info",
                      <i
                        className="icon-exchange"
/>,
                    ],
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": Array [
                    "More Info",
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "className": "icon-exchange",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "i",
                    },
                  ],
                  "type": "button",
                },
              ],
              "type": "div",
            },
          ],
          "type": "article",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="top-section"
>
                <div
                                className="top-row"
                >
                                <h1 />
                                <p
                                                className="rating"
                                />
                </div>
                <p />
</div>,
              <div
                className="bottom-container"
>
                <button
                                onClick={[Function]}
                >
                                Favorite
                                <i
                                                className="icon-star-empty"
                                />
                </button>
                <h1
                                className=""
                />
                <button
                                onClick={[Function]}
                >
                                More Info
                                <i
                                                className="icon-exchange"
                                />
                </button>
</div>,
            ],
            "className": "card-back face",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="top-row"
>
                    <h1 />
                    <p
                                        className="rating"
                    />
</div>,
                  <p />,
                ],
                "className": "top-section",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <h1 />,
                      <p
                        className="rating"
/>,
                    ],
                    "className": "top-row",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": undefined,
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "h1",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": undefined,
                        "className": "rating",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "p",
                    },
                  ],
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "p",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <button
                    onClick={[Function]}
>
                    Favorite
                    <i
                                        className="icon-star-empty"
                    />
</button>,
                  <h1
                    className=""
/>,
                  <button
                    onClick={[Function]}
>
                    More Info
                    <i
                                        className="icon-exchange"
                    />
</button>,
                ],
                "className": "bottom-container",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Favorite",
                      <i
                        className="icon-star-empty"
/>,
                    ],
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Favorite",
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "className": "icon-star-empty",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "i",
                    },
                  ],
                  "type": "button",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                    "className": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "h1",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "More Info",
                      <i
                        className="icon-exchange"
/>,
                    ],
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": Array [
                    "More Info",
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "className": "icon-exchange",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "i",
                    },
                  ],
                  "type": "button",
                },
              ],
              "type": "div",
            },
          ],
          "type": "article",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="inner-card-container"
>
          <article
                    className="card-front face"
          >
                    <img
                              alt="movie poster"
                              src="https://image.tmdb.org/t/p/w500undefined"
                    />
                    <div
                              className="bottom-container"
                    >
                              <button
                                        onClick={[Function]}
                              >
                                        Favorite
                                        <i
                                                  className="icon-star-empty"
                                        />
                              </button>
                              <h1
                                        className=""
                              />
                              <button
                                        onClick={[Function]}
                              >
                                        More Info
                                        <i
                                                  className="icon-exchange"
                                        />
                              </button>
                    </div>
          </article>
          <article
                    className="card-back face"
          >
                    <div
                              className="top-section"
                    >
                              <div
                                        className="top-row"
                              >
                                        <h1 />
                                        <p
                                                  className="rating"
                                        />
                              </div>
                              <p />
                    </div>
                    <div
                              className="bottom-container"
                    >
                              <button
                                        onClick={[Function]}
                              >
                                        Favorite
                                        <i
                                                  className="icon-star-empty"
                                        />
                              </button>
                              <h1
                                        className=""
                              />
                              <button
                                        onClick={[Function]}
                              >
                                        More Info
                                        <i
                                                  className="icon-exchange"
                                        />
                              </button>
                    </div>
          </article>
</div>,
        "className": "outer-card-container ",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <article
              className="card-front face"
>
              <img
                            alt="movie poster"
                            src="https://image.tmdb.org/t/p/w500undefined"
              />
              <div
                            className="bottom-container"
              >
                            <button
                                          onClick={[Function]}
                            >
                                          Favorite
                                          <i
                                                        className="icon-star-empty"
                                          />
                            </button>
                            <h1
                                          className=""
                            />
                            <button
                                          onClick={[Function]}
                            >
                                          More Info
                                          <i
                                                        className="icon-exchange"
                                          />
                            </button>
              </div>
</article>,
            <article
              className="card-back face"
>
              <div
                            className="top-section"
              >
                            <div
                                          className="top-row"
                            >
                                          <h1 />
                                          <p
                                                        className="rating"
                                          />
                            </div>
                            <p />
              </div>
              <div
                            className="bottom-container"
              >
                            <button
                                          onClick={[Function]}
                            >
                                          Favorite
                                          <i
                                                        className="icon-star-empty"
                                          />
                            </button>
                            <h1
                                          className=""
                            />
                            <button
                                          onClick={[Function]}
                            >
                                          More Info
                                          <i
                                                        className="icon-exchange"
                                          />
                            </button>
              </div>
</article>,
          ],
          "className": "inner-card-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <img
                  alt="movie poster"
                  src="https://image.tmdb.org/t/p/w500undefined"
/>,
                <div
                  className="bottom-container"
>
                  <button
                                    onClick={[Function]}
                  >
                                    Favorite
                                    <i
                                                      className="icon-star-empty"
                                    />
                  </button>
                  <h1
                                    className=""
                  />
                  <button
                                    onClick={[Function]}
                  >
                                    More Info
                                    <i
                                                      className="icon-exchange"
                                    />
                  </button>
</div>,
              ],
              "className": "card-front face",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "movie poster",
                  "src": "https://image.tmdb.org/t/p/w500undefined",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <button
                      onClick={[Function]}
>
                      Favorite
                      <i
                                            className="icon-star-empty"
                      />
</button>,
                    <h1
                      className=""
/>,
                    <button
                      onClick={[Function]}
>
                      More Info
                      <i
                                            className="icon-exchange"
                      />
</button>,
                  ],
                  "className": "bottom-container",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        "Favorite",
                        <i
                          className="icon-star-empty"
/>,
                      ],
                      "onClick": [Function],
                    },
                    "ref": null,
                    "rendered": Array [
                      "Favorite",
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "className": "icon-star-empty",
                        },
                        "ref": null,
                        "rendered": null,
                        "type": "i",
                      },
                    ],
                    "type": "button",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": undefined,
                      "className": "",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "h1",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        "More Info",
                        <i
                          className="icon-exchange"
/>,
                      ],
                      "onClick": [Function],
                    },
                    "ref": null,
                    "rendered": Array [
                      "More Info",
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "className": "icon-exchange",
                        },
                        "ref": null,
                        "rendered": null,
                        "type": "i",
                      },
                    ],
                    "type": "button",
                  },
                ],
                "type": "div",
              },
            ],
            "type": "article",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="top-section"
>
                  <div
                                    className="top-row"
                  >
                                    <h1 />
                                    <p
                                                      className="rating"
                                    />
                  </div>
                  <p />
</div>,
                <div
                  className="bottom-container"
>
                  <button
                                    onClick={[Function]}
                  >
                                    Favorite
                                    <i
                                                      className="icon-star-empty"
                                    />
                  </button>
                  <h1
                                    className=""
                  />
                  <button
                                    onClick={[Function]}
                  >
                                    More Info
                                    <i
                                                      className="icon-exchange"
                                    />
                  </button>
</div>,
              ],
              "className": "card-back face",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <div
                      className="top-row"
>
                      <h1 />
                      <p
                                            className="rating"
                      />
</div>,
                    <p />,
                  ],
                  "className": "top-section",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        <h1 />,
                        <p
                          className="rating"
/>,
                      ],
                      "className": "top-row",
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": undefined,
                        },
                        "ref": null,
                        "rendered": null,
                        "type": "h1",
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": undefined,
                          "className": "rating",
                        },
                        "ref": null,
                        "rendered": null,
                        "type": "p",
                      },
                    ],
                    "type": "div",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": undefined,
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "p",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <button
                      onClick={[Function]}
>
                      Favorite
                      <i
                                            className="icon-star-empty"
                      />
</button>,
                    <h1
                      className=""
/>,
                    <button
                      onClick={[Function]}
>
                      More Info
                      <i
                                            className="icon-exchange"
                      />
</button>,
                  ],
                  "className": "bottom-container",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        "Favorite",
                        <i
                          className="icon-star-empty"
/>,
                      ],
                      "onClick": [Function],
                    },
                    "ref": null,
                    "rendered": Array [
                      "Favorite",
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "className": "icon-star-empty",
                        },
                        "ref": null,
                        "rendered": null,
                        "type": "i",
                      },
                    ],
                    "type": "button",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": undefined,
                      "className": "",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "h1",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        "More Info",
                        <i
                          className="icon-exchange"
/>,
                      ],
                      "onClick": [Function],
                    },
                    "ref": null,
                    "rendered": Array [
                      "More Info",
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "className": "icon-exchange",
                        },
                        "ref": null,
                        "rendered": null,
                        "type": "i",
                      },
                    ],
                    "type": "button",
                  },
                ],
                "type": "div",
              },
            ],
            "type": "article",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
